tests/tensor/test_max_argmax.py

```diff
@@ -0,0 +1,693 @@
+import builtins
+
+import numpy as np
+import pytest
+
+from pytensor.compile.function import function
+from pytensor.compile.mode import get_default_mode
+from pytensor.compile.sharedvalue import shared
+from pytensor.configdefaults import config
+from pytensor.gradient import grad, numeric_grad
+from pytensor.graph.replace import vectorize_node
+from pytensor.tensor.basic import (
+    as_tensor_variable,
+    constant,
+    get_underlying_scalar_constant_value,
+)
+from pytensor.tensor.math import (
+    Argmax,
+    TensorMax,
+    argmax,
+    argmin,
+    max,
+    max_and_argmax,
+    min,
+)
+from pytensor.tensor.type import (
+    matrix,
+    tensor,
+)
+from pytensor.tensor.type_other import NoneConst
+from tests import unittest_tools as utt
+from tests.tensor.utils import (
+    eval_outputs,
+    random,
+)
+
+
+if config.mode == "FAST_COMPILE":
+    mode_opt = "FAST_RUN"
+else:
+    mode_opt = get_default_mode()
+
+
+class TestMaxAndArgmax:
+    def setup_method(self):
+        TensorMax.debug = 0
+
+    def test_basic(self):
+        # dbt: for some reason, Argmax does not work when I pass: n = as_tensor_variable(5.0)
```

Comment:
For some reason, Argmax does not work when I pass: ```n = as_tensor_variable(5.0)```. MaxandArgmax used to work fine.

Reply:
Hmm, what does numpy do with scalar arrays? We should do the same as them

Reply:
numpy seems to work fine

Reply:
The problem seems to be this line, which is forcing Axis to be a tuple, when it should be allowed to be `None` for the scalar case: https://github.com/pymc-devs/pytensor/blob/27bd9aaffb2aa4c3e8577c097a2d4cffe3e21cf2/pytensor/tensor/math.py#L362

Should be self.axis = axis

Reply:
Tried this and it does not seem to work.
This is the line in particular that causes a problem:
https://github.com/pymc-devs/pytensor/blob/e48ff56010f42ef4e656ddbfe253b4ffc93f500c/pytensor/link/c/basic.py#L1745-L1774

The ```if failure``` block never executes in case of non scalar inputs. 

https://github.com/pymc-devs/pytensor/blob/e48ff56010f42ef4e656ddbfe253b4ffc93f500c/pytensor/link/c/cutils.py#L102-L103

```run_cthunk``` is imported using this but I cannot find ```cutils_ext```.

Reply:
You may be hitting a windows/installation issue then. Does the test fail in the CI here as well?

Reply:
Yes the CI error is the same as what I face locally. The difference between a scalar like ```as_tensor_variable(5.0)``` and non-scalar like ```as_tensor_variable([5.0])``` is that in the former case, the ``` if failure:``` block executes and in case of non-scalars, it never does.